prover/src/logup_gkr/mod.rs

```diff
@@ -114,45 +119,72 @@ impl<E: FieldElement> EvaluatedCircuit<E> {
         let periodic_values = evaluator.build_periodic_values();
 
         let mut input_layer_wires =
-            Vec::with_capacity(main_trace.main_segment().num_rows() * num_fractions);
-        let mut main_frame = EvaluationFrame::new(main_trace.main_segment().num_cols());
-
-        let mut query = vec![E::BaseField::ZERO; evaluator.get_oracles().len()];
-        let mut periodic_values_row = vec![E::BaseField::ZERO; periodic_values.num_columns()];
-        let mut numerators = vec![E::ZERO; num_fractions];
-        let mut denominators = vec![E::ZERO; num_fractions];
-        for i in 0..main_trace.main_segment().num_rows() {
-            let wires_from_trace_row = {
-                main_trace.read_main_frame(i, &mut main_frame);
-                periodic_values.fill_periodic_values_at(i, &mut periodic_values_row);
-                evaluator.build_query(&main_frame, &mut query);
-
-                evaluator.evaluate_query(
-                    &query,
-                    &periodic_values_row,
-                    log_up_randomness,
-                    &mut numerators,
-                    &mut denominators,
-                );
-                let input_gates_values: Vec<CircuitWire<E>> = numerators
-                    .iter()
-                    .zip(denominators.iter())
-                    .map(|(numerator, denominator)| CircuitWire::new(*numerator, *denominator))
-                    .collect();
-                input_gates_values
-            };
-
-            input_layer_wires.extend(wires_from_trace_row);
-        }
+            unsafe { utils::uninit_vector(main_trace.main_segment().num_rows() * num_fractions) };
```

Comment:
A couple of nits:

- We should probably call the variable just `trace`; `main_trace.main_segment()` looks a bit weird.
- We can import `uninit_vector()` from utils at the top of the file.

Reply:
Done

Reply:
I would still import `uninit_vector` on line 10 above and not use `utils::` here.

Reply:
I haven't pushed yet :)

Reply:
Now it should be fixed